core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala

```diff
@@ -166,31 +167,44 @@ class OpWorkflowModelReader(val workflowOpt: Option[OpWorkflow]) extends MLReade
     features.filter(f => resultIds.contains(f.uid))
   }
 
-  private def resolveBlacklist
+  private def resolveBlocklist
   (
     json: JValue,
     wfOpt: Option[OpWorkflow],
     features: Array[OPFeature],
     path: String
   ): Try[Array[OPFeature]] = {
-    if ((json \ BlacklistedFeaturesUids.entryName) != JNothing) { // for backwards compatibility
-      for {
-        feats <- wfOpt
-          .map(wf => Success(wf.getAllFeatures() ++ wf.getBlacklist()))
-          .getOrElse(loadStages(json, BlacklistedStages, path).map(_._2))
-        allFeatures = features ++ feats
-        blacklistIds = (json \ BlacklistedFeaturesUids.entryName).extract[Array[String]]
-      } yield blacklistIds.flatMap(uid => allFeatures.find(_.uid == uid))
-    } else {
-      Success(Array.empty[OPFeature])
-    }
+    // For backward compatibility. The relevant field name is determined
+    // by the max length of the blocklist found for each name.
+    val potentialNames = Seq(
+      (json \ BlocklistedFeaturesUids.entryName, BlocklistedStages),
+      (json \ OldBlocklistedFeaturesUids.entryName, OldBlocklistedStages)
+    )
+    potentialNames.map { names =>
+      if (names._1 != JNothing) { // for backwards compatibility
+        for {
+          feats <- wfOpt
+            .map(wf => Success(wf.getAllFeatures() ++ wf.getBlocklist()))
+            .getOrElse(loadStages(json, names._2, path).map(_._2))
+          allFeatures = features ++ feats
+          blocklistIds = names._1.extract[Array[String]]
+        } yield blocklistIds.flatMap(uid => allFeatures.find(_.uid == uid))
+      } else {
+        Success(Array.empty[OPFeature])
+      }
+    }.maxBy(_.getOrElse(Array()).length)
   }
 
-  private def resolveBlacklistMapKeys(json: JValue): Try[Map[String, Set[String]]] = Try {
-    (json \ BlacklistedMapKeys.entryName).extractOpt[Map[String, List[String]]] match {
-      case Some(blackMapKeys) => blackMapKeys.map { case (k, vs) => k -> vs.toSet }
-      case None => Map.empty
-    }
+  private def resolveBlocklistMapKeys(json: JValue): Try[Map[String, Set[String]]] = Try {
+    // For backward compatibility. The relevant field name is determined
```

Comment:
5c: you have `extractOpt`, does it not return `Option[Map[...]]`?

Reply:
Indeed. Using `flatMap` now.